src/hotspot/share/sanitizers/address.hpp

```diff
@@ -57,7 +57,11 @@
 #ifdef ADDRESS_SANITIZER
 // ASAN_POISON_MEMORY_REGION is defined in <sanitizer/asan_interface.h>
 // ASAN_UNPOISON_MEMORY_REGION is defined in <sanitizer/asan_interface.h>
+#define ASAN_ONLY(code) code
+#define NOT_ASAN(code)
 #else
+#define ASAN_ONLY(code)
+#define NOT_ASAN(code)  code
 #define ASAN_POISON_MEMORY_REGION(addr, size) \
```

Comment:
This preprocessing directive is for the macros `ASAN_POISON_MEMORY_REGION` and `ASAN_UNPOISON_MEMORY_REGION`. You can read the comment before `#ifdef ADDRESS_SANITIZER`. So it is good to place your code at another macro `#ifdef ADDRESS_SANITIZER`.

Reply:
Okay, done.

Reply:
> You can read the comment before `#ifdef ADDRESS_SANITIZER`. So it is good to place your code at another macro `#ifdef ADDRESS_SANITIZER`.

The comment I meant is shown below, so I suggested that you put your code at another new macro.

```
// ASAN_POISON_MEMORY_REGION()/ASAN_UNPOISON_MEMORY_REGION()
//
// Poisons/unpoisons the specified memory region. When ASan is available this is the macro of the
// same name from <sanitizer/asan_interface.h>. When ASan is not available this macro is a NOOP
// which preserves the arguments, ensuring they still compile, but ensures they are stripped due to
// being unreachable. This helps ensure developers do not accidently break ASan builds.
```

But from the newest code, which uses only `INCLUDE_ASAN` instead of `ASAN_ONLY` and `NOT_ASAN`, I think you can use the exiting `ADDRESS_SANITIZER` instead of adding `INCLUDE_ASAN`.